cleanup/scripts/cleanup.sh

```diff
@@ -140,4 +148,70 @@ for SUFFIX in ${SUFFIX_ARR[@]}; do
     echo "   key pair ${KEY_NAME} deleted"
   done
 
+done
+
+echo "[Step 4] Prepare to delete long-running resources without owner:"
+ALL_INSTANCES=$(aws ec2 describe-instances --query 'Reservations[].Instances[?!not_null(Tags[?Key == `Owner`].Value)] | []' | jq '.[] | select(.State.Name != "terminated") | {LaunchTime: .LaunchTime, InstanceId: .InstanceId, Tags: .Tags}' | jq -c)
+INSTANCE_IDS=()
+for INSTANCE in ${ALL_INSTANCES[@]}; do
+  INSTANCE_ID=$(echo "${INSTANCE}" | jq '.InstanceId' | tr -d '"')
+  echo " * Instance ${INSTANCE_ID} ==>"
+  LAUNCH_TIME=$(echo "${INSTANCE}" | jq '.LaunchTime' | tr -d '"')
+  TIMESTAMP=$(date -D "%Y-%m-%dT%H:%M:%S+00:00" -d "${LAUNCH_TIME}" +%s)
+  TIME_DIFF=$((CURRENT_TIMESTAMP-TIMESTAMP))
+  echo "   Launch Time: ${LAUNCH_TIME} (${TIMESTAMP}), Diff: ${TIME_DIFF}"
+  if [[ $TIME_DIFF -gt $THRESHOLD_IN_SEC ]]; then INSTANCE_IDS+=("$INSTANCE_ID"); fi
+done
+for INSTANCE_ID in ${INSTANCE_IDS}
+do
+  aws ec2 terminate-instances --instance-ids "${INSTANCE_ID}"
+  echo "   instance ${INSTANCE_ID} shutting-down"
+done
+while [[ -n $(aws ec2 describe-instances --instance-ids "${INSTANCE_ID}" | jq '.Reservations[].Instances[].State.Name' | grep -v "terminated") ]]; do
+  echo "Wait for instances terminated ..."
+  sleep 5s
+done
```

Comment:
Just curious if the while loop only wait the last `${INSTANCE_ID}` in above for loop to terminated? (and the same question at the similar part of next section) thank you.

Reply:
Refined. Thank you!